gemcalibration/include/gem/calib/Calibration.h

```diff
@@ -0,0 +1,240 @@
+          throw (xgi::exception::Exception);
+
+        void resumeAction(xgi::Input *in, xgi::Output *out)
+          throw (xgi::exception::Exception);
+
+        void pauseAction(xgi::Input *in, xgi::Output *out)
+          throw (xgi::exception::Exception);
+	
+        void applyAction(xgi::Input *in, xgi::Output *out)
+          throw (xgi::exception::Exception);
+        
+        void setCalType(xgi::Input *in, xgi::Output *out)
+          throw (xgi::exception::Exception);
+
+        std::vector<Calibration*> v_gemcal;
+
+        calType_t m_calType;
+
+        std::map<calType_t, std::map<std::string, uint32_t>> m_scanParams{
+            {GBTPHASE  ,{{"nSamples",100},{"trigType", 0},}},
+            {LATENCY,{
+                {"nSamples"  , 100},
+                {"trigType"  , 0},
+                {"l1aTime"   , 250},
+                {"calPhase"  , 0},
+                {"mspl"      , 4},
+                {"scanMin"   , 0},
+                {"scanMax"   , 255},
+                {"vfatChMin" , 0},
+                {"vfatChMax" , 127},
+                {"vt2"       , 0},
+		{"trigThrottle"  ,100},
+                {"signalSourceType"       , 0},
+                }},
+            {SCURVE,{
+                {"nSamples"  , 100},
+		{"trigType"  , 0}, // TODO: TTC local should be only possible one
+                {"l1aTime"   , 250},
+                {"pulseDelay", 40},
+                {"latency"   , 33},
+                {"vfatChMin" , 0},
+                {"vfatChMax" , 127},
+                {"calPhase"  , 0},
+                }},
+	    {SBITARMDACSCAN  ,{
+		{"nSamples", 100},
+		{"comparatorType",0},
+		{"perChannelType",0},
+		{"vfatChMin" , 0},
+		{"vfatChMax" , 127},
+		}},
+	    {ARMDACSCAN  ,{
+		{"nSamples"  , 100},
+		{"trigType"  , 0},
+		{"vfatChMin" , 0},
+		{"vfatChMax" , 127},
+		}},
+            {TRIMDAC  , {
+		{"nSamples"   , 100}, 
+		{"trigType"   , 0}, // TODO: TTC local should be only possible one
+		{"nSamples"   , 100},
+		{"l1aTime"    , 250},
+                {"pulseDelay" , 40},
+                {"latency"    , 33},
+		{"mspl"       , 4},
+		{"trimValues" , 3},// TODO: need to be implemented properly in the back end in order to get a given number of points {-63,0,63}
+                                   // TODO: need to implement interaction with DB to get proper configurations per ARM DAC
+		}},
+	    {DACSCANV3  ,{
+	       {"nSamples",100},
+	       {"adcType",0},
+	      }},
+	    {CALIBRATEARMDAC,{
+	       {"nSamples"  , 100},
+	       {"trigType"  , 0}, // TODO: TTC local should be only possible one
+               {"l1aTime"   , 250},
+               {"pulseDelay", 40},
+               {"latency"   , 33},
+               {"calPhase"  , 0},
+		 // TODO: need to take the list of te ARM dac from a file
+	       }}  
+        };
+
+	struct  scanParamsRadioSelector{
+	  //std::string name;
+	  
+	  std::string label;
+	  std::vector<std::string>  radio_options;
```

Comment:
is `radio_` necessary?
the struct type already labels it such

Reply:
changed to simply `options`.